packages/ui/src/memberships/modals/BuyMembershipModal/BuyMembershipFormModal.tsx

```diff
@@ -177,19 +177,53 @@ export const BuyMembershipForm = ({
     }
   }, [data?.membershipsConnection.totalCount])
 
-  const isFormValid = !isUploading && form.formState.isValid && (!isValidator || validatorAccounts?.length)
+  const isFormValid =
+    !isUploading &&
+    form.formState.isValid &&
+    (!isValidator ||
+      (validatorAccountCandidates?.filter((account) => !account || !isValidValidatorAccount(account)).length === 0 &&
+        validatorAccountCandidates?.filter((account) => !!account).length))
   const isDisabled =
     type === 'onBoarding' && process.env.REACT_APP_CAPTCHA_SITE_KEY ? !captchaToken || !isFormValid : !isFormValid
 
   const addValidatorAccount = () => {
-    if (validatorAccountCandidate && isValidValidatorAccount) {
-      setValidatorAccounts([...new Set([...validatorAccounts, validatorAccountCandidate])])
-      form?.setValue('validatorAccountCandidate' as keyof MemberFormFields, undefined)
-    }
+    setValidatorAccounts([...(validatorAccountCandidates ?? []), undefined])
+    form?.setValue('validatorAccountCandidate' as keyof MemberFormFields, undefined)
   }
 
   const removeValidatorAccount = (index: number) => {
-    setValidatorAccounts([...validatorAccounts.slice(0, index), ...validatorAccounts.slice(index + 1)])
+    validatorAccountCandidates &&
+      setValidatorAccounts([
+        ...validatorAccountCandidates.slice(0, index),
+        ...validatorAccountCandidates.slice(index + 1),
+      ])
```

Comment:
So here maybe you can use `toSpliced`:
```suggestion
    setValidatorAccounts((accounts) => accounts.toSpliced(index, 1))
```